simpleast-core/src/test/java/com/discord/simpleast/code/JavaScriptRuleTest.kt

```diff
@@ -0,0 +1,225 @@
+package com.discord.simpleast.code
+
+import com.discord.simpleast.assertNodeContents
+import com.discord.simpleast.core.node.Node
+import com.discord.simpleast.core.node.StyleNode
+import com.discord.simpleast.core.parser.Parser
+import com.discord.simpleast.core.simple.SimpleMarkdownRules
+import com.discord.simpleast.core.utils.TreeMatcher
+import org.junit.Before
+import org.junit.Test
+
+
+class JavaScriptRulesTest {
+
+  private class TestState
+
+  private lateinit var parser: Parser<TestRenderContext, Node<TestRenderContext>, TestState>
+  private lateinit var treeMatcher: TreeMatcher
+
+  @Before
+  fun setup() {
+    val codeStyleProviders = CodeStyleProviders<TestRenderContext>()
+    parser = Parser()
+    parser
+        .addRule(CodeRules.createCodeRule(
+            codeStyleProviders.defaultStyleProvider,
+            CodeRules.createCodeLanguageMap(codeStyleProviders))
+        )
+        .addRules(SimpleMarkdownRules.createSimpleMarkdownRules())
+    treeMatcher = TreeMatcher()
+    treeMatcher.registerDefaultMatchers()
+  }
+
+  @Test
+  fun comments() {
+    val ast = parser.parse("""
+      ```js
+      /** Multiline
+          Comment
+      */
+      foo.bar(); // Inlined
+      // Line comment
+      
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+          /** Multiline
+              Comment
+          */
+        """.trimIndent(),
+        "// Inlined",
+        "// Line comment")
+  }
+
+  @Test
+  fun strings() {
+    val ast = parser.parse("""
+      ```js
+      const x = 'Hello';
+      const y = "world";
+      console.log(`Hi`);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "'Hello'",
+        "\"world\"",
+        "`Hi`")
+  }
+  @Test
+  fun stringsMultiline() {
+    val ast = parser.parse("""
+      ```js
+      const text = `
+      hello
+      world
+      `;
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "`",
+        """
+          hello
+          world
+        """.trimIndent(),
+        "`"
+    )
+  }
+
+  @Test
+  fun functions() {
+    val ast = parser.parse("""
+      ```js
+      function test(T) {
+        // Implementation
+      }
+      const fn = function() {};
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<JavaScript.FunctionNode<*>>(
+      """function test(T) {
+           // Implementation
+         }""".trimIndent(),
+      "function() {}")
```

Comment:
```suggestion
      "function test(T)",
      "function ()")
```
Reflect changes to function pattern